xesmf/backend.py

```diff
@@ -443,12 +547,30 @@ def esmf_regrid_finalize(regrid):
     regrid.destroy()
     regrid.srcfield.destroy()
     regrid.dstfield.destroy()
-    regrid.srcfield.grid.destroy()
-    regrid.dstfield.grid.destroy()
+    # regrid.srcfield.grid.destroy()
```

Comment:
what is the reason for commenting out these lines?
ESMF is prone to memory leaks if objects are not destroyed properly

Reply:
With the new structure, we have the `BaseRegridder` object that takes ESMF objects as input. Thus, destroying them after the weight computation could cause issues for a user who created those themself.  
More specifically, I commented out these lines because `SpatialAverager` uses the same grid in two different `BaseRegridder` instances, so I instead of recreating it, I decided to reuse the same object.
A partial solution could be to destroy the grids where appropriate in `Regridder` and `SpatialAverager`.

Reply:
ok let's see if this causes problems down the line. I will put an issue in to keep track of the potential problem